src/get-started/java-binding.md

```diff
@@ -491,93 +498,297 @@ as specified in [How to Run a Service section](#how-to-run-a-service).
+
+### Time Oracle Testing
+
+The TestKit allows to use [Time Oracle][time-oracle] in integration tests if
+your service depends on it. To do that, TestKit should be created with
+[`TimeProvider`][testkit-time-provider].
+Its implementation [`FakeTimeProvider`][testkit-fake-time-provider] mocks the
+source of external data (current time) and therefore allows to manually
+manipulate time that is returned by time service. Note that it only supports
+time in UTC.
+
+??? note "Test example"
+    ```java
+    @Test
+    void timeOracleTest() {
+      ZonedDateTime initialTime = ZonedDateTime.now(ZoneOffset.UTC);
+      FakeTimeProvider timeProvider = FakeTimeProvider.create(initialTime);
+      try (TestKit testKit = TestKit.builder()
+          .withService(MyServiceModule.class)
+          .withTimeService(timeProvider)
+          .build()) {
+        // Create an empty block
+        testKit.createBlock();
+        // The time service have submitted its first transaction in `afterCommit`
+        // method, but it has not been executed yet
+        Optional<ZonedDateTime> consolidatedTime1 = getConsolidatedTime(testKit);
+        // No time available till the time service transaction is processed
+        assertThat(consolidatedTime1).isEmpty();
+
+        // Advance the time
+        ZonedDateTime time1 = initialTime.plusSeconds(1);
+        timeProvider.setTime(time1);
+        testKit.createBlock();
+        // The time service has submitted its second transaction. The first must
+        // have been executed, with consolidated time now available and equal to
+        // initialTime
+        Optional<ZonedDateTime> consolidatedTime2 = getConsolidatedTime(testKit);
+        assertThat(consolidatedTime2).hasValue(initialTime);
+
+        // Advance the time
+        ZonedDateTime time2 = initialTime.plusSeconds(1);
+        timeProvider.setTime(time2);
+        testKit.createBlock();
+        // The time service have submitted its third transaction, and processed the
+        // second. The consolidated time must be equal to time1
+        Optional<ZonedDateTime> consolidatedTime3 = getConsolidatedTime(testKit);
+        assertThat(consolidatedTime3).hasValue(time1);
       }
     }
+
+    private Optional<ZonedDateTime> getConsolidatedTime(TestKit testKit) {
+      return testKit.applySnapshot(s -> {
+        TimeSchema timeSchema = TimeSchema.newInstance(s);
+        return timeSchema.getTime().toOptional();
+      });
+    }
     ```
 
-### Transactions
+### TestKit JUnit 5 extension
 
-To test transactions execution, you can use `MemoryDb`, as in the previous
-section.
+TestKit JUnit 5 extension simplifies writing tests that use TestKit. It allows to
+inject TestKit objects into test cases as a parameter and delete them
+afterwards. To enable it, define a [`TestKitExtension`][testkit-extension]
+object annotated with [`@RegisterExtension`][junit-register-extension] and
+provided with a builder that would be used to construct injected TestKit
```

Comment:
provided with a builder. The builder would be used to construct the injected TestKit objects: